api/v1beta1/azuremachine_webhook.go

```diff
@@ -154,13 +154,6 @@ func (mw *azureMachineWebhook) ValidateUpdate(ctx context.Context, oldObj, newOb
 		allErrs = append(allErrs, err)
 	}
 
-	if err := webhookutils.ValidateImmutable(
```

Comment:
should the new `AcceleratedNetworking` field be immutable? when it's not nil

Reply:
I think this is a fair change. We are removing an immutability-validation on deprecated field. I think we should have caught this and removed validation on `m.Spec.AcceleratedNetworking` while merging multiple network interfaces PR.